natlas-server/app/__init__.py

```diff
@@ -39,13 +39,78 @@ def unauthorized():
 		return redirect(url_for('main.index'))
 
 
-def create_app(config_class=Config, load_config=False):
+def load_natlas_config(app):
+	if not db.engine.has_table("config_item"):
+		return
+
+	updated_items = 0
+	from app.models import ConfigItem
+	# Look to see if any new config items were added that aren't currently in db
+	default_configs = config.get_defaults()
+	for key, item in default_configs:
+		conf_item = ConfigItem.query.filter_by(name=key).first()
+		if not conf_item:
+			conf_item = ConfigItem(name=key, type=item['type'], value=item['default'])
+			db.session.add(conf_item)
+			updated_items += 1
+		app.config[conf_item.name] = config.casted_value(conf_item.type, conf_item.value)
+	if updated_items > 0:
+		db.session.commit()
```

Comment:
This seems more like something that would fall in a DB migration script instead of on application startup.a

Reply:
Additionally, it seems like we're creating separate config tables for each "thing" Another common pattern is to create a generic configuration table that you can store any key value pair. I did something like this in [CertManager](https://github.com/ajacques/CertManager/tree/master/app/models/settings) by having specific classes that wrapped up each type of thing, but then stored into the same physical DB table.

Reply:
I agree that this can probably be done in a migration instead. I propose we table that for a future PR, and we can discuss the key value pair benefits for the other config types. The ConfigItem is a generic key value pair table already, however different types of configs have their own table because mentally I was preparing for a multi-tenant system where different tenants would have different agent configs, natlas services, agent scripts, etc.

Reply:
Captured the move to migration in https://github.com/natlas/natlas/issues/298